l1-contracts/contracts/bridge/L1NativeTokenVault.sol

```diff
@@ -0,0 +1,310 @@
+/// @dev Vault holding L1 native ETH and ERC20 tokens bridged into the hyperchains.
+/// @dev Designed for use with a proxy for upgradability.
+contract L1NativeTokenVault is
+    IL1NativeTokenVault,
+    IL1AssetHandler,
+    ReentrancyGuard,
+    Ownable2StepUpgradeable,
+    PausableUpgradeable
+{
+    using SafeERC20 for IERC20;
+
+    /// @dev The address of the WETH token on L1.
+    address public immutable override L1_WETH_TOKEN;
+
+    /// @dev L1 Shared Bridge smart contract that handles communication with its counterparts on L2s
+    IL1SharedBridge public immutable override L1_SHARED_BRIDGE;
+
+    /// @dev Era's chainID
+    uint256 public immutable ERA_CHAIN_ID;
+
+    /// @dev Indicates whether the hyperbridging is enabled for a given chain.
+    // slither-disable-next-line uninitialized-state
+    mapping(uint256 chainId => bool enabled) public hyperbridgingEnabled;
+
+    /// @dev Maps token balances for each chain to prevent unauthorized spending across hyperchains.
+    /// This serves as a security measure until hyperbridging is implemented.
+    /// NOTE: this function may be removed in the future, don't rely on it!
+    mapping(uint256 chainId => mapping(address l1Token => uint256 balance)) public chainBalance;
+
+    /// @dev A mapping assetId => tokenAddress
+    mapping(bytes32 assetId => address tokenAddress) public tokenAddress;
+
+    /// @notice Checks that the message sender is the bridgehub.
+    modifier onlyBridge() {
+        require(msg.sender == address(L1_SHARED_BRIDGE), "NTV not ShB");
+        _;
+    }
+
+    /// @notice Checks that the message sender is the bridgehub.
+    modifier onlyOwnerOrBridge() {
+        require((msg.sender == address(L1_SHARED_BRIDGE) || (msg.sender == owner())), "NTV not ShB or o");
+        _;
+    }
+
+    /// @notice Checks that the message sender is the shared bridge itself.
+    modifier onlySelf() {
+        require(msg.sender == address(this), "NTV only");
+        _;
+    }
+
+    /// @dev Contract is expected to be used as proxy implementation.
+    /// @dev Initialize the implementation to prevent Parity hack.
+    constructor(
+        address _l1WethAddress,
+        IL1SharedBridge _l1SharedBridge,
+        uint256 _eraChainId
+    ) reentrancyGuardInitializer {
+        _disableInitializers();
+        L1_WETH_TOKEN = _l1WethAddress;
+        ERA_CHAIN_ID = _eraChainId;
+        L1_SHARED_BRIDGE = _l1SharedBridge;
+    }
+
+    /// @dev Initializes a contract for later use. Expected to be used in the proxy
+    /// @param _owner Address which can change L2 token implementation and upgrade the bridge
+    /// implementation. The owner is the Governor and separate from the ProxyAdmin from now on, so that the Governor can call the bridge.
+    function initialize(address _owner) external reentrancyGuardInitializer initializer {
+        require(_owner != address(0), "NTV owner 0");
+        _transferOwnership(_owner);
+    }
```

Comment:
Do we really need to keep this contract ownable? Owner has no additional power anyway

Reply:
What can be added later should be added later

Reply:
uh, I see the point now